### PR TITLE
Mux connection cleanup for AMQP redesign

### PR DESCRIPTION
**Pull request by davilu**
## Checklist
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

### CODE REVIEW

**Comment by abhipsaMisra** [comment]
Build success: https://azure-iot-sdks.visualstudio.com/azure-iot-sdks/_build/results?buildId=11145

**Comment by abhipsaMisra** [comment]
Build started (with version change): https://azure-iot-sdks.visualstudio.com/azure-iot-sdks/_build/results?buildId=11150

**Comment by davilu** [comment]
Build with version change success: https://azure-iot-sdks.visualstudio.com/azure-iot-sdks/_build/results?buildId=11150

File: iothub/device/src/Transport/AmqpClientConnection.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ internal abstract class AmqpClientConnection
 
         internal TlsTransportSettings tlsTransportSettings { get; private set; }
 
-        public AmqpConnection amqpConnection { get; protected set; }
+        public AmqpConnection AmqpConnection { get; protected set; }
```
Comment (prmathur-microsoft) [inline iothub/device/src/Transport/AmqpClientConnection.cs]: I don't follow this coding standard. Shouldn't it be camelCase for member name?

File: iothub/device/src/Transport/AmqpClientConnectionPool.cs [hunk 2]
```diff
@@ -5,181 +5,117 @@ namespace Microsoft.Azure.Devices.Client.Transport
...
     {
-        ConcurrentDictionary<DeviceClientEndpointIdentity ,AmqpClientConnection> deviceConnectionDictionary;
-        ConcurrentBag<AmqpClientConnection> connectionPool;
-
-        ConcurrentDictionary<DeviceClientEndpointIdentity, AmqpClientConnection> deviceConnectionDictionaryIoTHubSas;
-        ConcurrentBag<AmqpClientConnection> connectionPoolIotHubSas;
-
-        AmqpClientConnectionFactory amqpClientConnectionFactory;
+        private static readonly TimeSpan TimeWait = TimeSpan.FromSeconds(10);
+        private ISet<AmqpClientConnectionMux> AmqpClientSasConnections;
```
Comment (prmathur-microsoft) [inline iothub/device/src/Transport/AmqpClientConnectionPool.cs]: Can you define magic number somewhere.

Comment (davilu) [reply iothub/device/src/Transport/AmqpClientConnectionPool.cs]: `TimeWait` is already `static` and it's specialized for `AmpqClientConnectionPool`. I believe this is the best place to put this constant.

File: iothub/device/src/Transport/AmqpClientConnection.cs (hunk 1 above)
Comment (davilu) [reply iothub/device/src/Transport/AmqpClientConnection.cs]: I'm not sure. There are a lot of classes using upper case, like `MethodInvokeRequest`, `MethodRequest` etc.

File: iothub/device/src/Microsoft.Azure.Devices.Client.csproj [hunk 3]
```diff
@@ -22,7 +22,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <Version>1.19.0</Version>
+    <Version>1.19.1</Version>
```
Comment (abhipsaMisra) [inline iothub/device/src/Microsoft.Azure.Devices.Client.csproj]: Agree about the version update. We should only update the versions.csv file, and leave the csproj to be updated during release

File: iothub/device/src/Transport/AmqpClientConnection.cs (hunk 1 above)
Comment (abhipsaMisra) [reply iothub/device/src/Transport/AmqpClientConnection.cs]: Property naming should follow PascalCasing

**Comment by abhipsaMisra** [comment]
We should also make sure that all modified/added classes and methods have logging added

**Comment by abhipsaMisra** [comment]
Coding style reference doc: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions

**Pull request closed without merging**